sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -335,7 +337,7 @@ class RelationalGroupedDataset protected[sql](
    * @since 1.6.0
    */
   def pivot(pivotColumn: String, values: Seq[Any]): RelationalGroupedDataset = {
-    pivot(Column(pivotColumn), values)
+    pivot(Column(pivotColumn), values.map(lit))
```

Comment:
This is going to allow `pivot(String, Seq[Any])` also take `Column`. Did I misread the codes?

Reply:
Yes, you did. This "old" interface only takes in a single named column (say, "a", but not "a+1") by its name, but we turn it into a `Column` just to reuse the same implementation.

Reply:
So, we did allow only liternals but not generic columns before, right?

Reply:
Yes, with Seq[Any] we only allow literal values, not `Column`s.

Reply:
Hm? I think we should allow it and then make this https://github.com/apache/spark/pull/22030#discussion_r208456164 assumption stay true.